src/multipole.F90

```diff
@@ -0,0 +1,184 @@
+    integer :: is_fissionable
+
+    associate (nuc => nuclides(i_table))
+
+      ! Open file for reading and move into the /isotope group
+      file_id = file_open(filename, 'r', parallel=.true.)
+      group_id = open_group(file_id, "/nuclide")
+
+      ! Load in all the array size scalars
+      call read_dataset(group_id, "length", multipole % length)
+      call read_dataset(group_id, "windows", multipole % windows)
+      call read_dataset(group_id, "num_l", multipole % num_l)
+      call read_dataset(group_id, "fit_order", multipole % fit_order)
+      call read_dataset(group_id, "max_w", multipole % max_w)
+      call read_dataset(group_id, "fissionable", is_fissionable)
+      if (is_fissionable == MP_FISS) then
+        multipole % fissionable = .true.
+      else
+        multipole % fissionable = .false.
+      end if
+      call read_dataset(group_id, "formalism", multipole % formalism)
+
+      call read_dataset(group_id, "spacing", multipole % spacing)
+      call read_dataset(group_id, "sqrtAWR", multipole % sqrtAWR)
+      call read_dataset(group_id, "start_E", multipole % start_E)
+      call read_dataset(group_id, "end_E", multipole % end_E)
+
+      ! Allocate the multipole array components
+      call multipole % allocate()
+
+      ! Read in arrays
+      call read_dataset(group_id, "data", multipole % data)
+      call read_dataset(group_id, "pseudo_K0RS", multipole % pseudo_k0RS)
+      call read_dataset(group_id, "l_value", multipole % l_value)
+      call read_dataset(group_id, "w_start", multipole % w_start)
+      call read_dataset(group_id, "w_end", multipole % w_end)
+      call read_dataset(group_id, "broaden_poly", multipole % broaden_poly)
+
+      call read_dataset(group_id, "curvefit", multipole % curvefit)
+
+      ! Delete ACE pointwise data
+      call read_dataset(group_id, "n_grid", nuc % n_grid)
+
+      deallocate(nuc % energy)
+      deallocate(nuc % total)
+      deallocate(nuc % elastic)
+      deallocate(nuc % fission)
+      deallocate(nuc % nu_fission)
+      deallocate(nuc % absorption)
+
+      allocate(nuc % energy(nuc % n_grid))
+      allocate(nuc % total(nuc % n_grid))
+      allocate(nuc % elastic(nuc % n_grid))
+      allocate(nuc % fission(nuc % n_grid))
+      allocate(nuc % nu_fission(nuc % n_grid))
+      allocate(nuc % absorption(nuc % n_grid))
+
+      nuc % total = ZERO
+      nuc % absorption = ZERO
+      nuc % fission = ZERO
+
+      ! Read in new energy axis (converting eV to MeV)
+      call read_dataset(group_id, "energy_points", nuc % energy)
+      nuc % energy = nuc % energy / 1.0D6
+
+      ! Get count and list of MT tables
+      call read_dataset(group_id, "MT_count", NMT)
+      allocate(MT(NMT))
+
+      call read_dataset(group_id, "MT_list", MT)
+
+      call close_group(group_id)
+
+      accumulated_fission = .false.
+
+      ! Loop over each MT entry and load it into a reaction.
+      do i = 1, NMT
+        write(MT_string, '(I3.3)') MT(i)
```

Comment:
You can use the `zero_padded` function from string.F90 here instead of an internal write.